Frends.MicrosoftSQL.ExecuteProcedure/Frends.MicrosoftSQL.ExecuteProcedure/ExecuteProcedure.cs

```diff
@@ -0,0 +1,184 @@
+                    }
+                }
+            }
+
+            if (options.SqlTransactionIsolationLevel is SqlTransactionIsolationLevel.None)
+                result = await ExecuteHandler(input, options, command, cancellationToken);
+            else
+            {
+                using var transaction = connection.BeginTransaction(GetIsolationLevel(options));
+                command.Transaction = transaction;
+                result = await ExecuteHandler(input, options, command, cancellationToken);
+            }
+
+            return result;
+        }
+        catch (Exception ex)
+        {
+            var eMsg = $"ExecuteProcedure exception: {ex}.";
+
+            if (options.ThrowErrorOnFailure)
+                throw new Exception(eMsg);
+
+            return new Result(false, 0, eMsg, null);
+        }
+        finally
+        {
+            SqlConnection.ClearAllPools();
+        }
+    }
+
+    private static async Task<Result> ExecuteHandler(Input input, Options options, SqlCommand command, CancellationToken cancellationToken)
+    {
+        Result result;
+        object dataObject;
+        SqlDataReader dataReader = null;
+        var table = new DataTable();
+
+        try
+        {
+            switch (input.ExecuteType)
+            {
+                case ExecuteTypes.NonQuery:
+                    dataObject = await command.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+                    result = new Result(true, (int)dataObject, null, JToken.FromObject(new { AffectedRows = dataObject }));
+                    break;
+                case ExecuteTypes.Scalar:
+                    dataObject = await command.ExecuteScalarAsync(cancellationToken).ConfigureAwait(false);
+                    result = new Result(true, 1, null, JToken.FromObject(new { Value = dataObject }));
+                    break;
+                case ExecuteTypes.ExecuteReader:
+                    dataReader = await command.ExecuteReaderAsync(cancellationToken).ConfigureAwait(false);
+                    table.Load(dataReader);
+                    result = new Result(true, dataReader.RecordsAffected, null, JToken.FromObject(table));
+                    await dataReader.CloseAsync();
+                    break;
+                default:
+                    throw new NotSupportedException();
+            }
+
+            if (command.Transaction != null)
+                await command.Transaction.CommitAsync(cancellationToken);
+
+            return result;
+        }
+        catch (Exception ex)
+        {
+            if (dataReader != null && !dataReader.IsClosed)
+                await dataReader.CloseAsync();
+
+            if (command.Transaction is null)
+            {
+                var eMsg = $"ExecuteHandler exception: 'Options.SqlTransactionIsolationLevel = None', so there was no transaction rollback. {ex}.";
+                if (options.ThrowErrorOnFailure)
+                    throw new Exception(eMsg);
```

Comment:
Include the original exception (`ex`) in the new Exception by passing it as inner exception param like this: `throw new Exception(eMsg)`